core/generator/generator.go

```diff
@@ -75,7 +75,16 @@ func Generate(
 		log.Fatal(ctx, log.KeyError, err)
 	}
 	if b != nil && (g.latestBlock == nil || b.Height == g.latestBlock.Height+1) {
-		s := state.Copy(g.latestSnapshot)
+		var s *state.Snapshot
+		if g.latestSnapshot == nil {
```

Comment:
I'm not 100% following why these nil snapshot changes should happen together with plumbing and checking the initial block hash in validation.